library/src/main/java/com/novoda/downloadmanager/lib/TarTruncator.java

```diff
@@ -31,31 +42,30 @@ public TarTruncator(DataWriter dataWriter) {
         }
     }
 
-    private boolean isNotFullOfZeroes(byte[] buffer) {
-        for (byte b : buffer) {
-            if (b != BYTE_ZERO) {
-                return true;
-            }
+    private int truncateEOFMarker(byte[] buffer, int length) {
```

Comment:
is this method more `findMarkerToTruncateEof()`. atm (truncate being a verb) it sounds like it will truncate the EOF marker and I don't think it is.